maturin drops env before invokin stuff so the CARGO_ envs are ignored
just hack around it for now until maybe maturin gets changed
--- a/Cargo.toml
+++ b/Cargo.toml
@@ -196,7 +196,8 @@
 #
 # [1]: https://github.com/astral-sh/ruff/pull/9031
 lto = "thin"
-codegen-units = 16
+codegen-units = 1
+panic = "abort"
 
 # Some crates don't change as much but benefit more from
 # more expensive optimization passes, so we selectively
